### ISSUE TITLE
Saving function evaluation using autodiff=false

### ISSUE THREAD

**Issue by gstein3m** [SciML/OrdinaryDiffEq.jl#1608]
When I monitor the function evaluations by 

global nr_fcn = 0

function ode_foo(du, u, p, t)
    global nr_fcn = nr_fcn + 1
    println(nr_fcn," ",t," ",u)
   du[1] = u[1] + t^2
end

prob = ODEProblem(ode_foo, [0.0], (0.0, 1.0))
sol = solve(prob, Rodas4(;autodiff=false,diff_type=Val{:forward}),dt=1.0, adaptive=false)
println(sol.destats)

I get the following results:

1 0.0 [0.0]
2 1.4901161193847656e-8 [0.0]
3 0.0 [0.0]
4 0.0 [0.0]
5 0.0 [1.4901161193847656e-8]
6 0.386 [1.917282740275065e-9]
7 0.21 [0.012699484443235735]
8 0.63 [0.15933273889151991]
9 1.0 [0.5480320046016063]
10 1.0 [0.40901870896937675]

Evaluations 1,6,7,8,9,10 are for the 6 stages of the method. Evaluations 2,3 for the time derivative and 4,5 for derivative with respect to u.
Evaluations 3,4 could be replaced by evaluation 1. Is it possible to implement this?

Best regards

Gerd

**Comment by ChrisRackauckas** [comment]
Yes. We would need to add a boolean to FiniteDiff.jl caches to tell it to assume that `fx` in the cache is correct, and thus not re-evaluate it. Then we would need to make sure that `k1` is the `fx`, and set that bool in the OrdinaryDiffEq cache construction, and then those should be eliminated.

**Comment by gstein3m** [comment]
The initialisation and usage of the caches is hard to understand for me. When I look into function calc_rosenbrock_differentiation! the first evaluation of f(x) is stored in cache.fsalfirst.
I now want to tell this FiniteDiff.gradients and FiniteDiff.Jacobians by its own caches.
In case of gradients this should be cache.grad_config. 
But the simple assignment cache.grad_config.fx  = cache.fsalfirst in function  calc_rosenbrock_differentiation! fails due to a type conflict?
